### ISSUE TITLE
m10k_parse failing?

### ISSUE THREAD

**Issue by pk400** [pk400/SPO600-Build-Framework#15]
Now that we have a couple parts up on the repo I was trying to test a few of them. I've hit a problem though and not sure what's wrong.

I have a .m10k folder on my $HOME directory, and inside there is _m10k_parse_ and _monkeys10k.config_, but when I run _./m10k cmake.tar.gz_ (for example) it returns with **./m10k: m10k_parse failed.** I'm thinking that _m10k_parse_ doesn't send the main the proper values, or it's not receiving them.

Any of the parsing guys know what's happening?

**Comment by miged** [comment]
Seems like the parser is having trouble reading the config file when it's being called from the main file. I'll get it fixed.

**Comment by pk400** [comment]
Alright I tried the updated parse, but still get the same results. When you run ./m10k on your machine does it get past the parse section?

**Comment by miged** [comment]
Yeah, I have it set up like yours and parsing ran fine after fixing it.

**Comment by pk400** [comment]
Alright it works now for some reason. Thanks for fixing it.

**Comment by pk400** [comment]
Actually I found out it only works when the files are located in the $~/.m10k directory. I had to move everything from the local repo directory to ~/.m10k and ran it from there to work.

**Comment by playhan** [comment]
I need your help.
I need to know what the exact example of this code below is in your code.

```
for p in reader:
    print '|'.join(p).rsplit(':', 1)[1]
```

if config file looks like this
group1:-option1,-option2
group2:-option5,-option3,-option2
group3:-option9

I guess that probably "," include in the parse file and each output start with "|".
I am sorry, I don't use python

**Comment by miged** [comment]
It basically takes away the group name and delimits the options with '|', like so:

```
-option1|-option2
-option5|-option3|-option2
-option9
```

**Comment by playhan** [comment]
thx, so there will be no problem.

**Comment by playhan** [comment]
Could you guys add empty line at the last line.
I need to recognize which STDIN is the last line.
Like this,

$$$$$$$$$$$$$$$$$$$$$$
-option1|-option2
-option5|-option3|-option2
-option9

$$$$$$$$$$$$$$$$$$$$$$

**Comment by pk400** [comment]
Okay I updated the monkeys10k.config to have an empty line ce427cddfe81790169f83cafd739681d5a1fe58f

**Comment by pk400** [comment]
Do you think you could update the parser to handle the following .config file:

```
group1:-fcx-fortran-rules -fcx-limited-range -ffinite-math-only -ffloat-store|-fno-cx-fortran-rules -fno-cx-limited-range -fno-finite-math-only -fno-float-store
group2:-falign-functions -falign-jumps -falign-labels -falign-loops -fcaller-saves|-fno-align-functions -fno-align-jumps -fno-align-labels -fno-align-loops -fno-caller-saves
group3:-fbranch-target-load-optimize -fbranch-target-load-optimize2|-fno-branch-target-load-optimize -fno-branch-target-load-optimize2
group4:-faggressive-loop-optimizations -fasynchronous-unwind-tables|-fno-aggressive-loop-optimizations -fno-asynchronous-unwind-tables
```

I believe the output is the same, just the input has off options also.

**Comment by miged** [comment]
Does every option in the config file have an off option? If so, I could just change the parser output to append the -fno options automatically, so you don't have to modify the original config.

**Comment by rav005** [comment]
@pk400 The parser will work fine you just need to replace "|"(pipe) with ","(coma). Because that's what we had before and in last class professor said we need options for on and off. The actual output suppose to be each line without "group #:" and piped "|" with off options.

**Comment by pk400** [comment]
So something like this?:

```
-fcx-fortran-rules -fcx-limited-range -ffinite-math-only -ffloat-store,-fno-cx-fortran-rules -fno-cx-limited-range -fno-finite-math-only -fno-float-store
-falign-functions -falign-jumps -falign-labels -falign-loops -fcaller-saves,-fno-align-functions -fno-align-jumps -fno-align-labels -fno-align-loops -fno-caller-saves
-fbranch-target-load-optimize -fbranch-target-load-optimize2,-fno-branch-target-load-optimize -fno-branch-target-load-optimize2
-faggressive-loop-optimizations -fasynchronous-unwind-tables,-fno-aggressive-loop-optimizations -fno-asynchronous-unwind-tables
```

**Comment by rav005** [comment]
Yea, all options that you want on followed by , then these options turned off.

**Comment by rav005** [comment]
Also, if we decide to modify parser to do that for us then we need to undo everything in config file. So, lets just keep what we have. I'll fix the large config file.

**Comment by miged** [comment]
I got some code written that appends the -fno- option for every -f option, and -f option for every -fno- option at the end the line after a delimiter. If that's what we want then I can send the pull request.

**Comment by rav005** [comment]
If you have that then don't upload the code, use that code with large config file and make new confige file to replace that large config file.
